lti_consumer/plugin/compat.py

```diff
@@ -94,3 +94,36 @@ def publish_grade(block, user, score, possible, only_if_higher=False, score_dele
         score_deleted=score_deleted,
         grader_response=comment,
     )
+
+
+def user_has_access(user, block, course_key):
```

Comment:
@shimulch In the merged deep linking implementation, I already make use of the `has_access` method. You'll have to rework it a bit to reuse the code.
Also, break this down into two smaller functions, each only running a small piece of LMS code. This is meant to be a compatibility layer, so no logic is to be done here.

Reply:
@giovannicimolin, I've refactored the ``compat.py`` and added a couple of smaller functions to keep logic out of it. Then added reusable functions in ``views.py`` so that we can make use of those functions inside different views. Let me know if you think the current approach is ok :-)